docs: v2 asset locks - automatic funding + bech32 addresses (#182)
dashpay/dips: dip-0027.md

* [DIP-0007: LLMQ Signing Requests / Sessions](https://github.com/dashpay/dips/blob/master/dip-0007.md)
* [DIP-0010: LLMQ InstantSend](https://github.com/dashpay/dips/blob/master/dip-0010.md)
* [DIP-0011: Identities](https://github.com/dashpay/dips/blob/master/dip-0011.md)

> * [DIP-0018: Dash Platform Payment Address Encodings](https://github.com/dashpay/dips/blob/master/dip-0018.md)

Reviewer: Let's include DIP-17 also

```suggestion
* [DIP-0017: Dash Platform Payment Addresses and HD Derivation](https://github.com/dashpay/dips/blob/master/dip-0017.md)
* [DIP-0018: Dash Platform Payment Address Encodings](https://github.com/dashpay/dips/blob/master/dip-0018.md)
```

* Its value is the amount of Dash to be locked
* The sum of all TxOut amounts must equal the value of the output

#### Version 2: Automatic Credit Assignment

> A `version = 2` Asset Lock transaction signals that each entry in `credit_outputs` MUST be automatically credited by Platform to the Identity matching the entry's public key hash or script hash, without an accompanying Identity Registration or Top-Up state transition.

Reviewer: This update is intended to allow sending directly to platform addresses, correct? The text mentions crediting to an identity, but platform addresses aren't linked to particular identities.

Reviewer: @knst can you check the comment above?

> A `version = 2` Asset Lock transaction signals that each entry in `credit_outputs` MUST be automatically credited by Platform to the Identity matching the entry's public key hash or script hash, without an accompanying Identity Registration or Top-Up state transition.

Reviewer: Maybe something like this. @PastaPastaPasta what do you think?

```suggestion
When Dash Platform implemented the Platform address system defined by [DIP-17](https://github.com/dashpay/dips/blob/master/dip-0017.md), there was no way to directly fund a Platform address from the Core chain. Version 2 Asset Lock transactions enable transferring Core chain funds to Platform addresses by signalling that each entry in `credit_outputs` MUST be automatically credited to the Platform address matching the entry's public key hash or script hash. No accompanying state transition is required.
```

> `version = 2` extends the set of allowed `credit_outputs[i].scriptPubKey` forms to `{P2PKH, P2SH}`; all other consensus rules are unchanged from `version = 1`.

Reviewer: ```suggestion
Version 2 extends the set of allowed `credit_outputs[i].scriptPubKey` forms to `{P2PKH, P2SH}`; all other consensus rules are unchanged from version 1.
```
